File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
int count(string s){
    int cnt{};
    for (int i = 0; i < s.size(); ++i) {
        cnt+=s[i]=='1';
    }
    return cnt;
}

    public:
int numberOfBeams(vector<string>& bank) {
    int cnt{},ans{};
    for (int i = 0; i < bank.size()-1; ++i) {
        cnt=count(bank[i]);
        int j = i+1;
        if(cnt){
            while (j<bank.size()&&!count(bank[j]) ){
                j++;
            }
            if(j==bank.size()){
                break;
            }
            ans+=count(bank[j])*cnt;
            i=j-1;
        }
    }
    return ans;
}
};

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cs
public class Solution {
	public  int NumberOfBeams(string[] bank)
	{
		List<int> list = new List<int>();
		for (int i = 0; i < bank.Length; i++)
		{
			int count = 0;
			for (int j = 0; j < bank[i].Length; j++)
			{
				count += (bank[i][j]=='1') ? 1 : 0;
			}
			if(count>0)
			{
				list.Add(count);
			}
		}
		int ans = 0;
		for(int i = 0;i < list.Count-1;i++)
		{
			ans += (list[i] * list[i + 1]);
		}
		return ans;
	}
}